pkg/controllers/dnspolicy/dns_helper.go

```diff
@@ -337,3 +264,84 @@ func createOrUpdateEndpoint(dnsName string, targets v1alpha1.Targets, recordType
 	endpoint.RecordTTL = recordTTL
 	return endpoint
 }
+
+// removeDNSForDeletedListeners remove any DNSRecords that are associated with listeners that no longer exist in this gateway
+func (r *dnsHelper) removeDNSForDeletedListeners(ctx context.Context, upstreamGateway *gatewayv1beta1.Gateway) error {
+	dnsList := &v1alpha1.DNSRecordList{}
+	//List all dns records that belong to this gateway
+	labelSelector := &client.MatchingLabels{
+		LabelGatewayReference: upstreamGateway.Name,
+	}
```

Comment:
using? DNSPolicyBackRefAnnotation it is weird that it is named annotation but used as label. We could change it to take the dnspolicy and use the label

Reply:
I just mean like this https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/pkg/controllers/dnspolicy/dnspolicy_dnsrecords.go#L124

Since you are adding namespace below, and we can only have one policy attached to the gateway it makes no real difference.

re, naming of consts, sure i guess we could create new ones with "label" in the name, but I'd imagine they just used those in the kuadrant-operator since they will always be the same value.